lib/components/Attachments/index.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useRef, useEffect, useState } from "react";
+
+import classnames from "classnames";
+import { Button, Typography, Toastr } from "neetoui";
+
+import { DIRECT_UPLOAD_ENDPOINT } from "common/constants";
+import useUppyUploader from "hooks/useUppyUploader";
+
+import AttachmentCard from "./AttachmentCard";
+import AttachmentProgressCard from "./AttachmentProgressCard";
+import { DEFAULT_UPPY_CONFIG, UPPY_UPLOAD_CONFIG } from "./constants";
+
+const Attachment = ({
+  endpoint = DIRECT_UPLOAD_ENDPOINT,
+  attachments = [],
+  className,
+  onChange,
+}) => {
+  const [fileProgresses, setFileProgresses] = useState([]);
+
+  const addFileRef = useRef(null);
+
+  const { uppy, isUploading } = useUppyUploader({
+    endpoint,
+    uppyConfig: DEFAULT_UPPY_CONFIG,
+    uppyUploadConfig: UPPY_UPLOAD_CONFIG,
+  });
+
+  const handleAddFile = event => {
+    const files = Array.from(event.target.files);
+    files.forEach(file => {
+      uppy.addFile({
+        name: file.name,
+        type: file.type,
+        data: file,
+      });
+    });
+
+    const newlyAddedFiles = [
+      ...uppy.getFiles().map(file => ({
+        filename: file.name,
+        signedId: "awaiting",
+        url: "",
+        progress: 0,
+      })),
+    ];
```

Comment:
Getting all the newly added files in this logic would be redundant. Directly map over the `files` variable.